### PR TITLE
refactor(crypto): cleanup `crypto.subtle.digest[Sync]()` logic

### PR DESCRIPTION
**Pull request by iuioiua**
This PR bumps test coverage of `crypto/crypto.ts` to 100% and aims to improve logic and readability.

This might interest you, @mbhrznr.

### CODE REVIEW

File: crypto/crypto.ts [hunk 1]
```diff
@@ -204,92 +194,60 @@ const stdCrypto: StdCrypto = ((x) => x)({
...
-        (WEB_CRYPTO_DIGEST_ALGORITHM_NAMES as readonly string[]).includes(
-          name,
-        ) &&
-        // and the data is a single buffer,
-        bytes
-      ) {
-        return webCrypto.subtle.digest(algorithm, bytes);
-      } else if (DIGEST_ALGORITHM_NAMES.includes(name as DigestAlgorithmName)) {
-        if (bytes) {
+      if (DIGEST_ALGORITHM_NAMES.includes(name as DigestAlgorithmName)) {
```
Comment (kt3k) [inline crypto/crypto.ts:197]: This seems to change the priority of the crypto implementation to be used.

Currently `digest` uses native Web Crypto `subtle.digest` if the given algorithm is one of 4 standard algorithms, but now it uses `stdCrypto.subtle.digestSync` (therefore `wasmCrypto.digest`). Is this change intentional?

Comment (iuioiua) [reply crypto/crypto.ts:213]: Made some tweaks now.

**Review by kt3k — APPROVED** [review]
LGTM
